app/src/main/java/com/medmapper/v33001/dto/Medicine.kt

```diff
@@ -6,10 +6,12 @@ import androidx.room.PrimaryKey
 
 @Entity(tableName = "Medicine")
 data class Medicine(
-    @PrimaryKey(autoGenerate = true) val medID: Int,
-    @ColumnInfo(name = "name") val name: String?,
+    @PrimaryKey(autoGenerate = true) var medID: Int,
+    @ColumnInfo(name = "name") val name: String,
     @ColumnInfo(name = "strength") val strength: String?,
     @ColumnInfo(name = "start date") val startDate: String?,
-    @ColumnInfo(name = "prescription length") val lengthInDays: Int
+    @ColumnInfo(name = "prescription length") val lengthInDays: Int?,
```

Comment:
Suggesting Nullable - this might not be filled out or an indefinite prescription

Reply:
Assume not nullable.  That is safer.

Only make something nullable if you know it will be nullable, and have proper handling around nullability throughout.